src/transformers/pipelines/question_answering.py

```diff
@@ -183,8 +183,14 @@ def __call__(self, *args, **kwargs):
         # Generic compatibility with sklearn and Keras
         # Batched data
         elif "X" in kwargs:
+            warnings.warn(
+                "Passing the `X` argument to the pipeline is deprecated and will be removed in v5.", FutureWarning
+            )
             inputs = kwargs["X"]
         elif "data" in kwargs:
+            warnings.warn(
+                "Passing the `data` argument to the pipeline is deprecated and will be removed in v5.", FutureWarning
+            )
```

Comment:
These should be actionable and specify which arguments should be passed instead

Reply:
Done!